components/content-service/pkg/archive/tar.go

```diff
@@ -69,9 +75,86 @@ func WithGIDMapping(mappings []IDMapping) BuildTarbalOption {
 	}
 }
 
+// ExtractTarbal extracts an OCI compatible tar file src to the folder dst, expecting the overlay whiteout format
+func ExtractTarbal(ctx context.Context, src io.Reader, dst string, opts ...TarOption) (err error) {
+	var cfg tarConfig
+	start := time.Now()
+	for _, opt := range opts {
+		opt(&cfg)
+	}
+
+	//nolint:staticcheck,ineffassign
+	span, ctx := opentracing.StartSpanFromContext(ctx, "extractTarbal")
+	span.LogKV("src", src, "dst", dst)
+	defer tracing.FinishSpan(span, &err)
+
+	re, wr := io.Pipe()
+	src = io.TeeReader(src, wr)
+	tarReader := tar.NewReader(re)
+	type Info struct {
+		UID, GID int
+	}
+	finished := make(chan bool)
+	m := make(map[string]Info)
+	go func() {
```

Comment:
nit: adding a `defer close(finished)` here makes the intent of `finish` clearer and prevents future versions of this code from inadvertently introducing code paths that don't close `finished`. You don't need to write to the channel for `<-finished` to work; reading from a closed channel returns the default value of its type (`false` in this case).